### ISSUE TITLE
Mac OSX 10.10

### ISSUE THREAD

**Issue by gabereiser** [Robmaister/SharpFont#54]
Having an issue with 3.0.1 on Mac OSX 10.10.  When instantiating the library it crashes out saying there's a bad parameter in the constructor for FFT_Init_FreeType in libfreetype.6.dylib

This is the library reference if I'm not mistaken so I think the issue is the wrapper trying to load libfreetype.6.dylib and call into it.  I tried on both x64 and x86 using Xamarin Studio.  My game engine kinda relies on this library right now for font rendering in opengl and would like to see this fixed however, realistically I need to make this a dependency on a tool to build texture fonts and not the engine itself.

**Comment by gabereiser** [comment]
I'll post the Application Output here in a bit.

**Comment by Robmaister** [comment]
I don't actually have a mac to test on, but I can definitely take a look at the error message and point you in the right direction or debug it through you. Other people have successfully used SharpFont on OS X.

I'm going to guess that this won't take too long to fix though. On a hunch, you are copying [SharpFont.dll.config](https://github.com/Robmaister/SharpFont/blob/master/Source/SharpFont/SharpFont.dll.config) to the output directory of your engine, right? If it's not in the same directory as SharpFont.dll when it gets loaded, it won't know where to look for freetype.

**Comment by gabereiser** [comment]
```ERROR - 7/23/2015 11:49:48 AM : libfreetype.6.dylib
ERROR - 7/23/2015 11:49:48 AM :   at (wrapper managed-to-native) SharpFont.FT:FT_Init_FreeType (intptr&)
  at SharpFont.Library..ctor () [0x00007] in <filename unknown>:0 ```
This is what I'm seeing on a x64 build...   I do have the SharpFont.dll.config copying to the output directory and it's pointing it's mapping for osx to libfreetype6.dylib

**Comment by gabereiser** [comment]
Turns out it's an x86 vs x64 problem, libfreetype.6.dylib is x86.  Compiling my own from source for x64 solved the problem.

**Comment by Robmaister** [comment]
That makes sense. If you'd like to contribute to that build so that other OS X users with x64 applications don't have to go through all of that, it would really be appreciated.

https://github.com/Robmaister/SharpFont.Dependencies

I can also add a line to the end of SharpFont.dll.config that overrides the dllmap on OS X for x64.

**Comment by HinTak** [comment]
I have been cross-compiling freetype for mac os x as fat binary (i.e. all three architectures -
i386 ppc x86_64 ) for a while now:
https://github.com/HinTak/Font-Validator/tree/master/bin/Darwin

btw, my freetype windows binaries (which are a bit newer than yours) are here
https://github.com/HinTak/Font-Validator/tree/master/bin/Win32
https://github.com/HinTak/Font-Validator/tree/master/bin/Win64

They aren't "statically linked" so you do need to do the relevant platform-magic to load all the dependencies in that directory. For windows, it is the one you use in the example. For macs os x, you either copy them to /usr/local/lib, or set DYLD_LIBRARY_PATH .

**Comment by Robmaister** [comment]
I have a collection of binaries here that you can contribute to: https://github.com/Robmaister/SharpFont.Dependencies/

For the Win64 binaries, make sure to apply win64.patch, otherwise it won't work with SharpFont on Windows x64 systems properly.

Right now for OS X, I'm just using the older copy of FreeType distributed with the OS and a .dll.config and [DllMap](http://www.mono-project.com/docs/advanced/pinvoke/dllmap/) to do the dynamic loading: https://github.com/Robmaister/SharpFont/blob/master/Source/SharpFont/SharpFont.dll.config

OS X builds would be appreciated, if you can figure out a way for the DllMap to prefer the SharpFont.Dependencies copy of the dylib and also fall back on the system-provided one, that would be very much appreciated.

**Comment by HinTak** [comment]
I have updated all the libfreetype binaries I bundled to a rather heavily patched 2.6.4 - as well as trying to put all the dependencies inside static as much as possible. The windows builds were a bit of a failure - harfbuzz itself brings in too many other things; but the mac os X one no longer depends on anything other than system libraries.

Please feel free to grab it. I wouldn't recommend falling back to system-provide one, as it tends to be extremely old.

My 2.6.4 being a bit heavily patched is a bit unfortunate - 2.6.4 itself was out with mac build broken, as well as some other issues; so 2.6.5 was out quite soon after; but I have already started my testing process and just chose to back-port only important patches, such as the mac-build fix etc. So while it is nominally 2.6.4, it has a few extra patches which bring it to a little bit beyond 2.6.5.
